### PR TITLE
[DeveloperGuide] Sort Command

### PR DESCRIPTION
**Pull request by Jo3LW**
1) Sort Feature
2) Sort Use case
3) Sort test case

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -656,6 +709,15 @@ testers are expected to do more *exploratory* testing.
...
    3. Other incorrect filter commands to try: `filter`, `filter all`, `filter 1`<br/>Expected: Similar to previous.
+
+### Sorting transactions from a client
+
+1. Sorting transactions.
+
+   1. Test case: `sort 1 latest`<br/>Expected: All transactions in index 1 client will be displayed in the transaction list panel, sorted with the latest transactions first.
+   If there are no transactions, the transaction list panel will be empty.
+   2. Test case: `sort 2 newest`<br/>Expected: No transactions for index 2 client is sorted. Error details shown in the `Application's Reply` panel.
+   3. Other incorrect sort commands to try: `sort all latest`, `sort oldest`, `sort newest`<br/>Expected: Similar to previous.
```
Comment (ElginL) [inline docs/DeveloperGuide.md]: what is the pre requisites to this? what if there are no clients in the list? Would sort 1 latest still show the expected output?

File: docs/diagrams/SortActivityDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+@startuml
```
Comment (ElginL) [inline docs/diagrams/SortActivityDiagram.puml:1]: This activity diagram has one extra weird arrow head. Look at the image generated

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -564,6 +606,17 @@ Users are able to perform several tasks within the application that is broken do
 
     Use case ends.
 
+#### **Use case: UC08 - Sort client by latest transaction** <br>
+**Preconditions: JeeqTracker has at least 1 client**
+
+**MSS**
+
+1. User requests to sort the client at index 1 by latest transactions
+2. JeeqTracker displays the list of transactions of client at index 1 sorted with the latest dates first.
```
Comment (ElginL) [inline docs/DeveloperGuide.md]: `sorted by` instead of `sorted with`?

**Review by ElginL — COMMENTED** [review]
Remember to sync, and add to the links at the top of the dg for the new things that you have added.

File: docs/diagrams/SortActivityDiagram.puml (hunk 2 above)
Comment (Jo3LW) [reply docs/diagrams/SortActivityDiagram.puml:1]: Great spot!

**Review by notle1706 — APPROVED** [review]
LGTM!

**Review by ElginL — APPROVED** [review]
LGTM

**Pull request merged**